cdap-ui/app/cdap/components/Experiments/DetailedView/AddModelToPipelineBtn/index.js

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {MyArtifactApi} from 'api/artifact';
+import {getCurrentNamespace} from 'services/NamespaceStore';
+import {isNilOrEmpty, objectQuery} from 'services/helpers';
+import getPipelineConfig from 'components/Experiments/DetailedView/AddModelToPipelineBtn/PipelineSkeleton';
+import {GLOBALS} from 'services/global-constants';
+import Popover from 'components/Popover';
+import IconSVG from 'components/IconSVG';
+import { connect } from 'react-redux';
+import uuidV4 from 'uuid/v4';
+import {myExperimentsApi} from 'api/experiments';
+import {createWorkspace, applyDirectives} from 'components/Experiments/store/CreateExperimentActionCreator';
+require('./AddModelToPipelineBtn.scss');
+
+const MMDS_PLUGINS_ARTIFACT_NAME = 'mmds-plugins';
+const ERROR_MSG = 'Unable to find plugins to create a scoring pipeline';
+
+class AddModelToPipelineBtn extends Component {
+  static propTypes = {
+    experimentId: PropTypes.string,
+    modelId: PropTypes.string,
+    modelName: PropTypes.string,
+    srcPath: PropTypes.string,
+    directives: PropTypes.arrayOf(PropTypes.string),
+    splitId: PropTypes.string
+  };
+
+  state = {
+    disabled: true,
+    error: null,
+    mmdsPluginsArtifact: null,
+    datapipelineArtifact: null,
+    wranglerArtifact: null,
+    corepluginsArtifact: null,
+    schema: null
+  };
+
+  cloneId = uuidV4();
+  batchPipelineUrl = window.getHydratorUrl({
+    stateName: 'hydrator.create',
+    stateParams: {
+      namespace: getCurrentNamespace(),
+      cloneId: this.cloneId,
+      artifactType: GLOBALS.etlDataPipeline
+    }
+  });
+
+  componentDidMount() {
+    this.setWorkspaceId();
+    this.fetchArtifactForPipelines();
+  }
+
+  setWorkspaceId = () => {
+    let workspaceId;
+    let {directives, srcPath} = this.props;
```

Comment:
These variables don't seem to be required to be defined here. Can be inside the callback function instead?